ArduCopter/mode_auto.cpp

```diff
@@ -1917,61 +1918,58 @@ bool ModeAuto::verify_payload_place()
         nav_payload_place.descend_start_altitude = inertial_nav.get_position_z_up_cm();
-        nav_payload_place.descend_throttle_level = 0;
+        nav_payload_place.descend_throttle_level = 1.0;
         nav_payload_place.state = PayloadPlaceStateType_Descending;
         FALLTHROUGH;
     case PayloadPlaceStateType_Descending:
-        // make sure we don't descend too far:
+        // make sure we don't descend too far
         debug("descended: %f cm (%f cm max)", (nav_payload_place.descend_start_altitude - inertial_nav.get_position_z_up_cm()), nav_payload_place.descend_max);
         if (!is_zero(nav_payload_place.descend_max) &&
             nav_payload_place.descend_start_altitude - inertial_nav.get_position_z_up_cm()  > nav_payload_place.descend_max) {
             nav_payload_place.state = PayloadPlaceStateType_Ascending_Start;
             gcs().send_text(MAV_SEVERITY_WARNING, "PayloadPlace: Reached maximum descent");
             return false; // we'll do any cleanups required next time through the loop
         }
-        // see if we've been descending long enough to calibrate a descend-throttle-level:
-        if (is_zero(nav_payload_place.descend_throttle_level) &&
-            now - nav_payload_place.descend_start_timestamp > descend_throttle_calibrate_time) {
-            // todo: Add filter as single sample is vulnerable to noise
-            nav_payload_place.descend_throttle_level = current_throttle_level;
+        // if the aircraft has been descending long enough, calibrate the decent thrust
+        if ((now - nav_payload_place.descend_start_timestamp > descend_throttle_calibrate_time - measure_time) && 
+            (now - nav_payload_place.descend_start_timestamp < descend_throttle_calibrate_time)) {
+            // measure decent thrust
+            nav_payload_place.descend_throttle_level = MIN(nav_payload_place.descend_throttle_level, current_throttle_level);
         }
-        // watch the throttle to determine whether the load has been placed
-        // debug("hover ratio: %f   descend ratio: %f\n", current_throttle_level/nav_payload_place.hover_throttle_level, ((nav_payload_place.descend_throttle_level == 0) ? -1.0f : current_throttle_level/nav_payload_place.descend_throttle_level));
-        if (current_throttle_level/nav_payload_place.hover_throttle_level > hover_throttle_placed_fraction &&
-            (is_zero(nav_payload_place.descend_throttle_level) ||
-             current_throttle_level/nav_payload_place.descend_throttle_level > descent_throttle_placed_fraction)) {
+        // check for reduced thrust requirement indicating possible payload touch down
+        if (current_throttle_level / nav_payload_place.hover_throttle_level > hover_throttle_placed_fraction &&
+            (now - nav_payload_place.descend_start_timestamp > descend_throttle_calibrate_time ||
+             current_throttle_level / nav_payload_place.descend_throttle_level > descent_throttle_placed_fraction)) {
```

Comment:
Existing issue, but more likely now. `hover_throttle_level` and `descend_throttle_level` could be zero and give us a FPE.

Reply:
I have moved the reference thrust level to the other side so it is multiplied not divided. This removes the risk of divide by zero while maintaining a conservative reference. If these checks fail then the land_complete flag will be used to move to the next step.